addons/base_vat/models/res_partner.py

```diff
@@ -820,6 +821,10 @@ def check_vat_de(self, vat):
         is_valid_stnr = stdnum.util.get_cc_module("de", "stnr").is_valid
         return is_valid_vat(vat) or is_valid_stnr(vat)
 
+    def check_vat_il(self, vat):
+        check_func = stdnum.util.get_cc_module('il', 'hp').is_valid if self.is_company else stdnum.util.get_cc_module('il', 'idnr').is_valid
```

Comment:
I don't get why we're doing that kind of stuff depending of if it's a company or not?
Couldn't we do the same than the fix in the module and always use `idnr`? :thinking: 

Is it for the check on the first digit?

Reply:
@FlorianGilbert The format is different if the contact is a company or not. The difference is only on the first digit, it must be a 5 for companies. 
We could use only `idnr` but it would be more accurate to select the right one depending on if it is a company or not. I agree it can be confusing for the users if the format is changing.

Tell me if I need to change and only use `idnr`.

Reply:
@FlorianGilbert @mathcoutant I don't believe it is correct to access `self.is_company` here, as the function is called on potentially multiple partners in https://github.com/odoo/odoo/blob/16.0/addons/base_vat/models/res_partner.py#L170

So either you'll need to override `_run_vat_test` or change the line above to work on `partner` and not `self`

Reply:
> @FlorianGilbert @mathcoutant I don't believe it is correct to access `self.is_company` here, as the function is called on potentially multiple partners in https://github.com/odoo/odoo/blob/16.0/addons/base_vat/models/res_partner.py#L170
> 
> So either you'll need to override `_run_vat_test` or change the line above to work on `partner` and not `self`

Indeed, you're right, @mathcoutant may I ask you to fix it?
Thank you :)

Reply:
@FlorianGilbert I have looked into the proposed changes and changing partner to self seem to work.

But I also noticed that I forgot to add `'is_company'` as constraint of `check_vat` with this PR, to check when only `is_company` is changed. But this change causes problem when changing both vat and is_company so I changed the write method to ensure that the check after the change of is_company is called after the one of vat. There may be a better way to do this...
There is the new PR: https://github.com/odoo/odoo/pull/174817

Can I have your opinion on this? Maybe the best solution is to only use `idnr` and remove the specific check for the company...

Reply:
Hello @mathcoutant 
As my first opinion on this PR was to only use idnr, let's change the code for that. I would prefer avoid adding a new constraint, etc etc.
Thank you, feel free to assign me as reviewer :wink: